Source/Core/Core/AchievementManager.cpp

```diff
@@ -1460,22 +1462,27 @@ void AchievementManager::HandleAchievementTriggeredEvent(const rc_runtime_event_
                   (Config::Get(Config::RA_BADGES_ENABLED)) ?
                       DecodeBadgeToOSDIcon(it->second.unlocked_badge.badge) :
                       nullptr);
-  PointSpread spread = TallyScore();
-  if (spread.hard_points == spread.total_points)
-  {
-    OSD::AddMessage(
-        fmt::format("Congratulations! {} has mastered {}", m_display_name, m_game_data.title),
-        OSD::Duration::VERY_LONG, OSD::Color::YELLOW,
-        (Config::Get(Config::RA_BADGES_ENABLED)) ? DecodeBadgeToOSDIcon(m_game_badge.badge) :
-                                                   nullptr);
-  }
-  else if (spread.hard_points + spread.soft_points == spread.total_points)
+  if (m_game_data.achievements[game_data_index].category == RC_ACHIEVEMENT_CATEGORY_CORE)
   {
-    OSD::AddMessage(
-        fmt::format("Congratulations! {} has completed {}", m_display_name, m_game_data.title),
-        OSD::Duration::VERY_LONG, OSD::Color::CYAN,
-        (Config::Get(Config::RA_BADGES_ENABLED)) ? DecodeBadgeToOSDIcon(m_game_badge.badge) :
-                                                   nullptr);
+    PointSpread spread = TallyScore();
+    if (spread.hard_points == spread.total_points &&
+        it->second.remote_unlock_status != UnlockStatus::UnlockType::HARDCORE)
+    {
+      OSD::AddMessage(
+          fmt::format("Congratulations! {} has mastered {}", m_display_name, m_game_data.title),
+          OSD::Duration::VERY_LONG, OSD::Color::YELLOW,
+          (Config::Get(Config::RA_BADGES_ENABLED)) ? DecodeBadgeToOSDIcon(m_game_badge.badge) :
+                                                     nullptr);
+    }
+    else if (spread.hard_points + spread.soft_points == spread.total_points &&
+             it->second.remote_unlock_status == UnlockStatus::UnlockType::LOCKED)
```

Comment:
Sidenote, is this `remote_unlock_status` ever updated within a session? It seems to me as if this should be updated when the achievement is awarded but I don't see it, maybe I'm just missing it?

Reply:
It currently isn't updated in the middle of a session and has no need to be - if the current session has an unlock then that's all the information the AchievementManager currently needs to know. There's a future update in the works (criticality unknown but will probably be getting there before full release anyways) to retry unlock requests if they fail for connection reasons that will likely be using this.